### ISSUE TITLE
remove @BaseUrl and @DefaultHeaders decorators

### ISSUE THREAD

**Issue by mjerez-radical** [Paldom/angular2-rest#18]
This should be configurable at execution time

**Comment by auvipy** [comment]
any proposed solutions?

**Comment by M-jerez** [comment]
Yes, what i mean here is that using variables instead decorators might be a better approach, much more flexible and code would not need to recompile if URL or any header needs to be changed.

**Comment by 0xE282B0** [comment]
@mjerez-radical what do you think about passing a function as @BaseUrl parameter like in PR #22 ?
